code/modules/mob/living/simple_animal/hostile/terror_spiders/terror_spiders.dm

```diff
@@ -0,0 +1,826 @@
+	if (stat != DEAD)
+		if (degenerate > 0)
+			adjustToxLoss(rand(1,10))
+		if (regen_points < regen_points_max)
+			regen_points += regen_points_per_tick
+		if ((bruteloss > 0) || (fireloss > 0))
+			if (regen_points > regen_points_per_hp)
+				if (bruteloss > 0)
+					adjustBruteLoss(-1)
+					regen_points -= regen_points_per_hp
+				else if (fireloss > 0)
+					adjustFireLoss(-1)
+					regen_points -= regen_points_per_hp
+		if (prob(5))
+			CheckFaction()
+	if (stat == DEAD)
+		if (prob(2))
+			// 2% chance every cycle to decompose
+			visible_message("<span class='notice'>\the dead body of the [src] decomposes!</span>")
+			gib()
+	..()
+
+
+/mob/living/simple_animal/hostile/poison/terror_spider/death(gibbed)
+	if (!gibbed)
+		msg_terrorspiders("[src] has died in [get_area(src)].")
+		//if (!ckey && spider_tier < 3)
+		//	say(pick("Mistresssss will end you...", "Doom waitssss... for you...","She comessssss for your flesh..."))
+	if (!hasdroppedloot)
+		hasdroppedloot = 1
+		if (ts_count_dead == 0)
+			visible_message("<span class='userdanger'>The Terrors have awoken!</span>")
+		ts_count_dead++
+		ts_death_last = world.time
+		if (spider_awaymission)
+			ts_count_alive_awaymission--
+		else
+			ts_count_alive_station--
+	..()
+
+
+/mob/living/simple_animal/hostile/poison/terror_spider/handle_automated_action()
+	if(!stat && !ckey) // if we are not dead, and we're not player controlled
+		if(AIStatus != AI_OFF && !target)
+			var/my_ventcrawl_freq = freq_ventcrawl_idle
+			if (ts_count_dead > 0)
+				if (world.time < (ts_death_last + ts_death_window))
+					my_ventcrawl_freq = freq_ventcrawl_combat
+			if (cocoon_target)
+				if(get_dist(src, cocoon_target) <= 1)
+					spider_steps_taken = 0
+					DoWrap()
+				else
+					if (spider_steps_taken > spider_max_steps)
+						spider_steps_taken = 0
+						cocoon_target = null
+						busy = 0
+						stop_automated_movement = 0
+					else
+						spider_steps_taken++
+						CreatePath(cocoon_target)
+						step_to(src,cocoon_target)
+						if (spider_debug > 0)
+							visible_message("<span class='notice'>\the [src] moves towards [cocoon_target] to cocoon it.</span>")
+			else if (path_to_vent)
+				if (entry_vent)
+					if (spider_steps_taken > spider_max_steps)
+						path_to_vent = 0
+						stop_automated_movement = 0
+						spider_steps_taken = 0
+						path_to_vent = 0
+						entry_vent = null
+					else if (get_dist(src, entry_vent) <= 1)
+						path_to_vent = 0
+						stop_automated_movement = 1
+						spider_steps_taken = 0
+						spawn(50)
+							stop_automated_movement = 0
+						TSVentCrawlRandom(entry_vent)
+					else
+						spider_steps_taken++
+						CreatePath(entry_vent)
+						step_to(src,entry_vent)
+						if (spider_debug > 0)
+							visible_message("<span class='notice'>\the [src] moves towards the vent [entry_vent].</span>")
+				else
+					path_to_vent = 0
+			else if (istype(src,/mob/living/simple_animal/hostile/poison/terror_spider/purple) && prob(5))
```

Comment:
again with the istype(src) checks, make a `try_special_action()` or some shit proc for handle_automated_action to call and every class to overload